envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/MetadataNodeGroup.kt

```diff
@@ -64,11 +69,38 @@ class MetadataNodeGroup(val properties: SnapshotProperties) : NodeGroup<Group> {
         return ListenersHostPortConfig(ingressHost, ingressPort, egressHost, egressPort)
     }
 
+    private fun metadataToAccessLogFilterHttpCode(metadata: Struct): AccessLogFilterHttpCode? {
+        val httpFilterCode = metadata.fieldsMap["access_log_filter_http_code"]?.stringValue
+        var op: Op? = null
+        var code: String? = null
+
+        if (!httpFilterCode.isNullOrBlank()) {
+            val regex = """(eq|ge|le?):(\d+)""".toRegex()
```

Comment:
we can just use split by colon and eliminate the need for regex

Reply:
I can change it to simple split, but personally I prefer keep it. This is simple regex, even made it more simpler and strict in current PR. I think having regex allows us use less ifs to check pattern.